infrastructure/doc_processing.py

```diff
@@ -82,13 +83,31 @@ def generate_processing_doc():
                 info = param.tooltip_3liz
             else:
                 info = ''
+
+            option = ''
+            if param.defaultValue():
+                option += 'Default: {' + str(param.defaultValue()) + '} <br> '
+
+            if isinstance(param, QgsProcessingParameterNumber):
+                nameType = ''
+                if param.dataType() == 1:
```

Comment:
For better readability, you should always use the enum directly. Replace the `1` by `QgsProcessingParameterNumber.Double`